Dockerfiles/atg-jupyter-general/Dockerfile

```diff
@@ -1,78 +1,53 @@
-FROM jupyter/minimal-notebook
+# Use base image with the requested Python version 3.7
+FROM jupyter/minimal-notebook:29a003bb3030
 
 USER root
 
 RUN apt-get -y update
 RUN apt-get -y install libgl1-mesa-glx
 
-RUN conda install --quiet --yes \
-    'altair' \
-    'arviz' \
-    'bokeh' \
-    'beautifulsoup4' \
-    'control' \
-    'dash' \
-    'filterpy' \
-    'gensim' \
-    'holoviews' \
-    'ipytest' \
-    'IPython' \
-    'ipywidgets' \
-    'jupyterlab' \
-    'matplotlib' \
-    'mkl-service' \
-    'mesa' \
-    'mne' \
-    'mpmath' \
-    'networkx' \
-    'nltk' \
-    'numba' \
-    'numpy=1.18.5' \
-    'openpyxl' \
-    'opencv' \
-    'pandas' \
-    'peakutils' \
-    'pillow' \
-    'plotly' \
-    'powerlaw' \
-    'requests' \
-    'rpy2' \
-    'scikit-image' \
-    'scikit-learn' \
-    'scipy=1.4.1' \
-    'seaborn' \
-    'spacy' \
-    'statsmodels' \
-    'sympy' \
-    'thinkx' \
-    'xlrd' \
-    'xlsxwriter' \
-    && \
-    conda clean --al -f -y && \
-    # Activate ipywidgets extension in the environment that runs the notebook server
-    jupyter nbextension enable --py widgetsnbextension --sys-prefix && \
-    # Also activate ipywidgets extension for JupyterLab
-    # Check this URL for most recent compatibilities
-    # https://github.com/jupyter-widgets/ipywidgets/tree/master/packages/jupyterlab-manager
-    jupyter labextension install @jupyter-widgets/jupyterlab-manager@^2.0.0 --no-build && \
-    jupyter labextension install @bokeh/jupyter_bokeh@^2.0.0 --no-build && \
-    jupyter labextension install jupyter-matplotlib@^0.7.2 --no-build && \
-    jupyter lab build -y && \
-    jupyter lab clean -y && \
-    npm cache clean --force && \
-    rm -rf "/home/${NB_USER}/.cache/yarn" && \
-    rm -rf "/home/${NB_USER}/.node-gyp" && \
-    fix-permissions "${CONDA_DIR}" && \
-    fix-permissions "/home/${NB_USER}"
-    
-RUN pip install --upgrade pip && \
-    pip install --no-cache-dir gap-stat gym progressbar2 pygam stochastic tensorflow-gpu keras
+# Install dependencies of the pyjags package
+RUN apt-get -y install pkg-config jags
+
+# Install python packages using pip
+RUN pip install --upgrade pip
+RUN pip install --no-cache-dir tensorflow-gpu==2.0.0
+RUN pip install --no-cache-dir altair arviz bokeh beautifulsoup4 control
+RUN pip install --no-cache-dir dash filterpy gensim holoviews ipytest
+RUN pip install --no-cache-dir IPython ipywidgets jupyterlab matplotlib
+RUN pip install --no-cache-dir mesa mne mpmath networkx nltk
+RUN pip install --no-cache-dir numba numpy openpyxl pandas peakutils
+RUN pip install --no-cache-dir pillow plotly powerlaw pymc3 requests
+RUN pip install --no-cache-dir scikit-image scikit-learn scipy seaborn
+RUN pip install --no-cache-dir spacy statsmodels sympy thinkx xlrd xlsxwriter
+RUN pip install --no-cache-dir gap-stat gym progressbar2 pygam stochastic keras opencv-python pyjags csaps
```

Comment:
_Minor optimization_: It might be a good idea to consolidate these `RUN` statement into a single statement so that all the packages are installed and committed to the same layer (see also: [Dockerfile best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run)).

Reply:
It is a good idea. However, the intention here was to install a few packages at a time. Particularly when doing conda installs, I realized that installing a few at a time is faster than installing all at once. It also helps with faster troubleshooting. That is, when everything is on one install command, it can take very long before you the command errors out, say, due to package conflicts or missing dependencies. And that means it takes very long to iteratively fix things. So, I think let's leave as is for now. Thanks for the suggestion though.

Reply:
That makes sense @joshuagetega especially if there are issues with the package installs. I hadn't considered that aspect, as I was just thinking about the final image and not the process of building everything.